assets/global.js

```diff
@@ -846,15 +846,31 @@ class VariantSelects extends HTMLElement {
         const html = new DOMParser().parseFromString(responseText, 'text/html')
         const destination = document.getElementById(`price-${this.dataset.section}`);
         const source = html.getElementById(`price-${this.dataset.originalSection ? this.dataset.originalSection : this.dataset.section}`);
+
         if (source && destination) destination.innerHTML = source.innerHTML;
 
         const price = document.getElementById(`price-${this.dataset.section}`);
+        const sku = document.getElementById(`sku-${this.dataset.section}`);
 
         if (price) price.classList.remove('visibility-hidden');
+        if (sku) this.updateSKU(html);
         this.toggleAddButton(!this.currentVariant.available, window.variantStrings.soldOut);
       });
   }
 
+  updateSKU(html) {
+    const id = `sku-${this.dataset.section}`;
+    const destination = document.getElementById(id);
+    const source = html.getElementById(id);
+
+    if (source && destination) destination.innerHTML = source.innerHTML;
+    if (destination.innerHTML !== '#') {
+      destination.classList.remove('visually-hidden');
+    } else {
+      destination.classList.add('visually-hidden');
```

Comment:
Adding this class will hide it from sighted users but keep it in the DOM, so this will be read out by screen readers. To completely hide it, you need to either remove the paragraph from the DOM or give it an additional `aria-hidden="true"` attribute.

https://user-images.githubusercontent.com/22390758/192733908-d0a1dcbd-f47b-41e0-9fed-b6d07742a055.mov

Reply:
@metamoni thanks for this opinion. I had spent a while thinking about it. I'm open to discuss and investigate it deeper. The reason why I decided not to remove it completely from DOM is to let screen readers know that there is supposed to be SKU even if it's empty now. I still assume that let's say first variant might not have SKU but the second one have it and I'd like to let users know about it from a first stem without switching to the next variant.

Reply:
There should be a way to remove/hide it when it's empty, and add/show it when we have a SKU. Having an empty `#` is irrelevant to both sighted and screen reader users, particularly as we're not clearly saying that this is a the SKU - it's just either a number or a symbol.